include/cutlass/gemm/collective/intel_pvc_mma.hpp

```diff
@@ -187,29 +189,66 @@ struct CollectiveMma<
 
     Tensor tAr_view = make_tensor(static_cast<decltype(tAr) &&>(tAr).data(),
                             Shape<Int<VecA>, Int<FragsM>, Int<FragsK>>{});
     Tensor tBr_view = make_tensor(static_cast<decltype(tBr) &&>(tBr).data(),
-                            Shape<Int<VecB>, Int<FragsN>, Int<FragsK>>{},
-                            Stride<_1, Int<get<1>(SubgroupTileShape{}) / 2>, Int<VecB>>{});
+                                  Shape<Int<VecB>, Int<FragsN>, Int<FragsK>>{},
+                                  Stride<_1, Int<VecB * FragsK>, Int<VecB>>{});
 
     // Instantiate the M MA object
     TiledMma tiled_mma;
 
+    int K = size<1>(mainloop.gmem_tiled_copy_a.tensor);
+
+    auto sub_group_id = ThreadIdxX() / SubgroupSize;
+    /* Cooperative prefetch
+       Divice the thread space to sg_per_wg_m x sg_per_wg_n, all the threads in one row/col use the same tile A/B. 
+       Each thread loads sizeof(tile A or B) / numof(sg_per_wg_n or sg_per_wg_m). 
+       
+       Currently, sg_per_wg_m x sg_per_wg_n = 4 x 8 is the most efficient
+    */
+    // TODO: Replace the demo cooperative prefetch with more general way.
+    Tensor tAi = make_tensor(
+        make_inttuple_iter(
+            *gA.data() +
+            make_coord((sub_group_id % sg_per_wg_n % 4) * get<0>(MmaAtomShape{}), 0)),
+        make_layout(make_shape(_1{}, _1{}, K),
+                    make_stride(_1{}, E<0>{}, E<1>{})));
+    Tensor tBi = make_tensor(
+        make_inttuple_iter(
+            *gB.data() +
+            make_coord((sub_group_id / sg_per_wg_n % 2 * 2) * get<1>(MmaAtomShape{}),
+                       (sub_group_id / sg_per_wg_n / 2 % 2) * get<2>(MmaAtomShape{}))),
+        make_layout(make_shape(_1{}, _1{}, K),
+                    make_stride(_1{}, E<0>{}, E<1>{})));
     //
     // Mainloop
     //
-   for (int k_tile = 0, k = 0; k_tile < k_tile_count; ++k_tile, k += get<2>(MmaAtomShape()) * FragsK)
-   {
-     // Copy gmem to rmem for the first k_tile
-     copy(mainloop.gmem_tiled_copy_a, gA(_,_,k), tAr);
-     copy(mainloop.gmem_tiled_copy_b, gB(_,_,k/2), tBr);
-
-     cute::gemm(tiled_mma, accum, tAr_view, tBr_view, src_accum);
-   }
+    int prefetch_k = 0;
+    for (int i = 0; i < 3; i++) {
+      prefetch(mainloop.gmem_tiled_copy_a, tAi(_, _, prefetch_k));
+      prefetch(mainloop.gmem_tiled_copy_b, tBi(_, _, prefetch_k));
+      prefetch_k += get<2>(SubgroupTileShape{});
+    }
+
+    for (int k_tile = 0, k = 0; k_tile < k_tile_count;
+         ++k_tile, k += get<2>(SubgroupTileShape{})) {
+      // Copy gmem to rmem for the first k_tile
+      copy(mainloop.gmem_tiled_copy_a, gA(_, _, k), tAr);
+      copy(mainloop.gmem_tiled_copy_b, gB(_, _, k), tBr);
+
+      prefetch(mainloop.gmem_tiled_copy_a, tAi(_, _, prefetch_k));
+      prefetch(mainloop.gmem_tiled_copy_b, tBi(_, _, prefetch_k));
+      prefetch_k += get<2>(SubgroupTileShape{});
```

Comment:
Wouldn't `prefetch_k` go out of bounds here?

Reply:
Prefetch instrinsics would do the boundary check internally.